ConfigSpace/hyperparameters.pyx

```diff
@@ -1395,10 +1863,324 @@ cdef class NormalIntegerHyperparameter(IntegerHyperparameter):
             return np.rint((self.upper - self.lower) / self.q) + 1
 
 
-cdef class CategoricalHyperparameter(Hyperparameter):
-    cdef public tuple choices
-    cdef public tuple probabilities
-    cdef public int num_choices
+cdef class BetaIntegerHyperparameter(IntegerHyperparameter):
+    cdef public alpha
+    cdef public beta
+    cdef public bfhp
+    cdef normalization_constant
+
+
+    def __init__(self, name: str, alpha: Union[int, float], beta: Union[int, float],
+                 default_value: Union[int, None] = None, q: Union[None, int] = None,
+                 log: bool = False,
+                 lower: Optional[int] = None,
+                 upper: Optional[int] = None,
+                 meta: Optional[Dict] = None) -> None:
+        r"""
+        An integer hyperparameter.
+
+        Its values are sampled from a normal distribution
+        :math:`\mathcal{N}(\mu, \sigma^2)`.
+
+        Example
+        -------
+
+            >>> import ConfigSpace as CS
+            >>> import ConfigSpace.hyperparameters as CSH
+            ...                                                 beta=1, log=False)
+            >>> cs.add_hyperparameter(beta_int_hp)
+            beta_int, Type: NormalInteger, Alpha: 3 Beta: 1, Default: 1
+
+        Parameters
+        ----------
+        name : str
+            Name of the hyperparameter with which it can be accessed
+        alpha : int, float
+            Alpha parameter of the distribution, from which hyperparameter is sampled
+        beta : int, float
+            Beta parameter of the distribution, from which
+            hyperparameter is sampled
+        default_value : int, optional
+            Sets the default value of a hyperparameter to a given value
+        q : int, optional
+            Quantization factor
+        log : bool, optional
+            If ``True``, the values of the hyperparameter will be sampled
+            on a logarithmic scale. Defaults to ``False``
+        lower : int, float, optional
+            Lower bound of a range of values from which the hyperparameter will be sampled
+        upper : int, float, optional
+            Upper bound of a range of values from which the hyperparameter will be sampled
```

Comment:
Same here with making these non-optional and looking through the class to remove error checking on it.

Reply:
fixed